src/fastlane/release_actions/lib/fastlane/plugin/release_actions/helper/version/version.rb

```diff
@@ -0,0 +1,162 @@
+require_relative 'acceptor'
+
+# Version represents a semantic version as defined by Semantic Versioning 2.0.0.
+# Semantic versions must include a normal version number (i.e. 1.0.0) and can
+# optionally include labels indicating pre-release version and/or build metadata.
+# Version can compare and bump segments of a semantic version.
+#
+# For more details on the specification, see: https://semver.org/.
+class Version
+  include Comparable
+
+  attr_reader :segments, :prerelease, :build
+
+  def initialize(version)
+    # Acceptor ensures our input version is valid per the semver spec. See
+    # `Version::Acceptor` for details.
+    unless Acceptor.new(version).valid?
```

Comment:
It seems strange to me that the acceptor parses through the string to check for validity, but then we use straight up pattern matching for the actual version parsing. Do we get a benefit from the pre-check? Wouldn't it make sense to use the state machine to actually parse the version while you're already going through it?

Reply:
Yep, I had the same thought. The way I think about it - Acceptor is just a lightweight verifier and ensures that the string is able to be parsed by my string munging later. I could store the parts and return a version from this I suppose. I'll think more about it.

Reply:
I'd prefer to keep Acceptor as a single-purpose DFA for now. We aren't being as efficient as possible, but this is just a personal preference for simple unitaskers. If you feel strongly, I can change my mind and do the potentially better thing now.